pkg/certificates/reconciler/certificates.go

```diff
@@ -95,16 +105,29 @@ func (r *reconciler) ReconcileKind(ctx context.Context, secret *corev1.Secret) p
 	cert, _, err := parseAndValidateSecret(secret, true)
 	if err != nil {
 		r.logger.Infof("Secret invalid: %v", err)
-		sans := []string{certificates.DataPlaneNamePrefix + secret.Namespace, certificates.LegacyFakeDnsName}
 		// Check the secret to reconcile type
-		var keyPair *certificates.KeyPair
-		if secret.Labels[r.secretTypeLabelName] == dataPlaneSecretType {
-			keyPair, err = certificates.CreateCert(ctx, caPk, caCert, expirationInterval, sans...)
-		} else if secret.Labels[r.secretTypeLabelName] == controlPlaneSecretType {
-			keyPair, err = certificates.CreateCert(ctx, caPk, caCert, expirationInterval, sans...)
-		} else {
+
+		routingId := secret.Labels[r.secretRoutingId]
+		if routingId == "" {
+			routingId = "0"
+		}
+
+		var sans []string
+		switch secret.Labels[r.secretTypeLabelName] {
+		case controlPlaneSecretType:
+			sans = []string{certificates.ControlPlaneName}
+		case dataPlaneRoutingSecretType:
+			sans = []string{certificates.DataPlaneRoutingName(routingId), certificates.LegacyFakeDnsName}
```

Comment:
I have not yet gotten the `routingId`. Why do we need multiple? Who/What would set that label? Based on what?

Reply:
It enables maintaining and distinguishing between multiple data plane pipelines (e.g. of multiple tenants)
Right now we pick it up from the secret.

Reply:
See [comment](https://github.com/knative-sandbox/control-protocol/pull/270#issuecomment-1493695477) below. I'm not sure we have finished the discussion about mTLS and multi-tenancy yet.